'Stop generation' button

tijszwinkels opened the pull request.

tijszwinkels committed:
'Stop generation' button

slush0 commented:
Any plans to merge this? Generation sometimes gets into cycle or the model generates obvious bogus and it would be handy to preserve the pool (and chat) resources for other tasks...

slush0 commented:
As discussed with @borzunov on Discord, the same effect without adding the "stop" command can be implemented by closing the websocket.

tijszwinkels commented:
That sounds a lot cleaner! I'll update this MR somewhere in the coming days.

tijszwinkels committed:
Simplified 'stop generating' functionality.
Instead of changing the websocket api, we just close the websocket
if we want to stop generating.

tijszwinkels commented:
I changed the 'stop generation' button to just close the websocket. This works like a charm. :)

borzunov commented on static/chat.js:
```diff
@@ -157,8 +164,10 @@ function receiveReplica(inputs) {
     } else {
-      $('.loading-animation, .speed, .suggest-join').remove();
+      $('.loading-animation, .speed, .suggest-join, .generation-controls').remove();
+      ws.close();
```
```suggestion
      resetSession();
```

This is a smarter function that closes the websocket only if it's still open and resets the generation state (so that we feed the prefix to the model if you continue talking to it).

borzunov commented:
Thanks for the PR! Once I have time, I'll check that everything works properly and merge this.

tijszwinkels committed:
ws close suggestion from @borzunov

Co-authored-by: Alexander Borzunov <hxrussia@gmail.com>

borzunov commented on static/chat.js:
```diff
@@ -146,7 +153,7 @@ function receiveReplica(inputs) {
     lastReplica.text(newText);
 
-    if (!response.stop) {
+    if (!response.stop && stop == false) {
```
```suggestion
    if (!response.stop && !stop) {
```

borzunov committed:
Update static/chat.js
